### PR TITLE
Tweak audio recording widget

### LINKED ISSUE #32: When recording fails, we get both an error message and a Yay success message
If I click on record with my microphone turned off, I get this:
image
and the Play, Download and Delete buttons appear.

If I click Play, nothing happens, and if I click Download I get a `.webm` file containing just `undefined`.

The correct behaviour would be to prevent the Yay! message from being displayed and to have just the Record button displayed, maybe by setting `this.audioControl.setValue(null);` like in `deleteRecording()`?

### LINKED ISSUE #88: Studio-W: while playing audio, there should be a stop button
When you play the audio you just recorded or uploaded in step 1, you can't stop it. The play button should transform into a stop button while it's playing the audio so you can interrupt it.

### LINKED ISSUE #89: Studio-W: warn user to listen to their recorded audio before continuing
When I try to record, especially the first time in a given session, I often have to try multiple times before it's right. Recording doesn't actually start right away half the time, so I record but the first few words are missing.

That's going to be hard to fix, but we should warn the user: after recording, two notes should appear between "re-record" and "play/download/delete":
 - "Before continuing, play back you audio to verify it recorded correctly."
 - "If your audio is good, download it in case you need to restart the app."

or maybe something shorter to that general effect.

### PR DESCRIPTION
**Pull request by joanise**
Several improvements:
 - remove false success message that sometimes happened
 - put a warning to prefer this only for short audio tests
 - while listening to the playback, display a stop button, and reset on end/error/stop

Fixes: #89
Fixes: https://github.com/ReadAlongs/Studio-Web/issues/32
Fixes: #88

### CODE REVIEW

File: packages/studio-web/src/app/upload/upload.component.html [hunk 1]
```diff
@@ -72,6 +72,7 @@ <h2 i18n="Title for upload box" class="subtitle">Select an audio file</h2>
                 <div *ngIf="inputMethod.audio === 'mic'" class="mb-0 mb-md-3">
                     <div class="row">
                         <h2 i18n="Title for record box" class="subtitle">Record your voice here</h2>
+                        <p i18n="Warning for record box">Best suited for short tests.</p>
```
Comment (roedoejet) [inline packages/studio-web/src/app/upload/upload.component.html]: I don't think this is the appropriate place for this. Maybe put it as a tooltip or as part of the shepherd tour. I find it confusing to have this as a `p` element above the button.

File: packages/studio-web/src/app/upload/upload.component.html [hunk 2]
```diff
@@ -84,18 +85,23 @@ <h2 i18n="Title for record box" class="subtitle">Record your voice here</h2>
                         Stop
                       </button>
                     </div>
+                  <p *ngIf="!recording && audioControl.value !== null" i18n="playback invitation">Please Play your audio to make sure it is OK, and save it for reuse if it is.</p>
```
Comment (roedoejet) [inline packages/studio-web/src/app/upload/upload.component.html]: Similarly here, I'd rather this was a temporary toast message than markup on the page.

File: packages/studio-web/src/app/upload/upload.component.html [hunk 3]
```diff
@@ -84,18 +85,23 @@ <h2 i18n="Title for record box" class="subtitle">Record your voice here</h2>
...
+                    <button i18n="Audio stop button" class="audioControl" color="warn" (click)="stopPlayback()" *ngIf="!recording && audioControl.value !== null && playing" mat-button aria-label="Stop button">
+                      <mat-icon class="mat-icon-lg">stop_circle</mat-icon>
+                      Stop
+                    </button>
                   </div>
                   <div class="col-12 col-lg-4">
-                    <button i18n="Audio download button" class="audioControl" color="primary" (click)="downloadRecording()" *ngIf="!recording && audioControl.value !== null" mat-button color="primary" aria-label="Audio download button">
-                      <mat-icon class="mat-icon-lg">file_download</mat-icon>
-                      Download
+                    <button i18n="Audio download button" class="audioControl" color="primary" (click)="downloadRecording()" *ngIf="!recording && audioControl.value !== null" mat-button color="primary" aria-label="Audio save button">
```
Comment (roedoejet) [inline packages/studio-web/src/app/upload/upload.component.html:100]: Yep, this is better wording!

File: packages/studio-web/src/app/upload/upload.component.ts [hunk 4]
```diff
@@ -140,16 +141,22 @@ export class UploadComponent implements OnInit {
   playRecording() {
     if (!this.playing && this.audioControl.value !== null) {
       let player = new window.Audio();
+      this.player = player
       player.src = URL.createObjectURL(this.audioControl.value);
-      player.onended = () => {
-        this.playing = false;
-      };
+      player.onended = () => this.stopPlayback();
```
Comment (roedoejet) [inline packages/studio-web/src/app/upload/upload.component.ts:152]: Nice! this is the proper way to do it.

**Review by roedoejet — CHANGES_REQUESTED** [review]
I think the messaging around the buttons should be made into toast messages and/or instructions in the shepherd tour.

File: packages/studio-web/src/app/upload/upload.component.html (hunk 2 above)
Comment (joanise) [reply packages/studio-web/src/app/upload/upload.component.html]: toastr message is a good idea, yes, I agree that would be better.

File: packages/studio-web/src/app/upload/upload.component.html (hunk 1 above)
Comment (dhdaines) [reply packages/studio-web/src/app/upload/upload.component.html]: I agree, it looks out of place

File: packages/studio-web/src/app/upload/upload.component.html (hunk 2 above)
Comment (dhdaines) [reply packages/studio-web/src/app/upload/upload.component.html]: Also "Play" maybe shouldn't be capitalized?

**Review by dhdaines — APPROVED** [review]
Looks good, Aidan already suggested some appropriate changes

File: packages/studio-web/src/app/upload/upload.component.html (hunk 1 above)
Comment (joanise) [reply packages/studio-web/src/app/upload/upload.component.html]: Yeah, I guess I'll take it out. I don't like it as a tooltip either:
image
and this is no better:
image

**Review by dhdaines — APPROVED** [review]

**Pull request merged**